react/components/MaybeAuth.tsx

```diff
@@ -2,7 +2,7 @@ import React, {PureComponent} from 'react'
 import Loading from './Loading'
 
 const LOGIN_PATH = '/login'
-const AUTH_STORE_URL = '/_v/private/authenticated/store'
+const API_SESSION_URL = '/api/sessions?items=*'
```

Comment:
You need to make a `POST /api/sessions` with a header `Content-Type: application/json` to initialize session . (retrieve `vtex_session` and `vtex_segment` cookies)

Reply:
PS: Check if you already have these cookies, if not make the request.

Reply:
Actually `vtex_session` is (or should be) an HTTP only cookie, so there is no way to know if there cookie is there using javascript.